### PR TITLE
feat: Universal link

### PR DESCRIPTION
**Pull request by Crash--**
Added support for Universal link for iOS

### CODE REVIEW

File: react/AppLinker/index.jsx [hunk 1]
```diff
@@ -107,19 +115,36 @@ export class AppLinker extends React.Component {
...
+  href: PropTypes.string.isRequired,
+  /*
+    Path used for "native link"
+  */
+  nativePath: PropTypes.string
 }
 
 export default AppLinker
-export { NATIVE_APP_INFOS }
+export { NATIVE_APP_INFOS, getUniversalLinkDomain, generateUniversalLink }
```
Comment (y-lohse) [inline react/AppLinker/index.jsx]: Why are we exporting the universal link helpers from this component ?

File: react/AppLinker/native.js [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
-    name: 'Cozy Drive'
-  },
-  banks: {
-    appId: isAndroidApp() ? 'io.cozy.banks.mobile' : 'io.cozy.banks',
-    uri: 'cozybanks://',
-    name: 'Cozy Banks'
-  }
+import { UNIVERSALLINK_URL } from 'cozy-ui/transpiled/react/AppLinker/native.config'
+export const getUniversalLinkDomain = () => {
+  return UNIVERSALLINK_URL
```
Comment (y-lohse) [inline react/AppLinker/native.js]: Is there a point to having a function and a constant exposing the same value?

File: react/AppLinker/native.config.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    uri: 'cozydrive://',
+    name: 'Cozy Drive'
+  },
+  banks: {
+    appId: isAndroidApp() ? 'io.cozy.banks.mobile' : 'io.cozy.banks',
+    uri: 'cozybanks://',
+    name: 'Cozy Banks'
+  }
+}
+export const UNIVERSALLINK_URL = 'https://links.mycozy.cloud'
```
Comment (y-lohse) [inline react/AppLinker/native.config.js]:
```suggestion
export const UNIVERSAL_LINK_URL = 'https://links.mycozy.cloud'
```

and the same for the other uses of the constant?

File: react/AppLinker/native.js [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
+}
+/*
+  slug string drive 
+  path : /path/to/view
+  fallbackUrl: string : https://...mcyozy.cloud
```
Comment (y-lohse) [inline react/AppLinker/native.js]:
```suggestion
  fallbackUrl: string : https://...mycozy.cloud
```

File: react/AppLinker/native.spec.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import { generateUniversalLink, getUniversalLinkDomain } from './native'
+describe('native functions', () => {
+  it('should generate the universalink', () => {
+    const universalLink = generateUniversalLink({
+      slug: 'drive',
+      nativePath: '/files/1',
+      fallbackUrl: 'https://drive.cozy.tools/files/1'
+    })
+    const endLink =
+      getUniversalLinkDomain() +
```
Comment (y-lohse) [inline react/AppLinker/native.spec.js]: I would suggest not using a tested function to generate the *expected* result. I think hardcoding the domain here would be ok, no?

File: react/AppLinker/index.jsx (hunk 1 above)
Comment (Crash--) [reply react/AppLinker/index.jsx]: It seems easier to do: 
import AppLinker, {getUniversalLinkDomain} from 'cozy-ui/AppLinker'  

than : 
import AppLinker from 'cozy-ui/AppLinker'  
import {getUniversalLinkDomain} from 'cozy-ui/Applinker/native'

File: react/AppLinker/native.js (hunk 2 above)
Comment (Crash--) [reply react/AppLinker/native.js]: I prefer to have a getter. It gives us more flexibility to change the behavior / add check if needed (like checking if the url ends with a '/' or not).

File: react/AppLinker/index.jsx (hunk 1 above)
Comment (y-lohse) [reply react/AppLinker/index.jsx]: But is `getUniversalLinkDomain ` going to be used outside of cozy-ui? If yes we can leave it as it is :)

File: react/AppLinker/native.js (hunk 2 above)
Comment (y-lohse) [reply react/AppLinker/native.js]: I don't mind the getter, but then wouldn't it be easier to not have the constant, or at least in the same file?

File: react/AppLinker/index.jsx (hunk 1 above)
Comment (Crash--) [reply react/AppLinker/index.jsx]: Yes, it will. (in the mobile handleDeeplink process)

File: react/AppLinker/native.js (hunk 2 above)
Comment (Crash--) [reply react/AppLinker/native.js]: We need to override the value :/

**Pull request merged**